include/sparrow/layout/variable_size_binary_array.hpp

```diff
@@ -20,31 +20,77 @@
 #include <numeric>
 #include <ranges>
 #include <string>
+#include <vector>
+#include <cstddef>
 
+#include "sparrow/arrow_interface/arrow_array.hpp"
+#include "sparrow/arrow_interface/arrow_schema.hpp"
 #include "sparrow/arrow_array_schema_proxy.hpp"
 #include "sparrow/buffer/buffer_adaptor.hpp"
+#include "sparrow/buffer/u8_buffer.hpp"
 #include "sparrow/layout/array_bitmap_base.hpp"
 #include "sparrow/layout/layout_iterator.hpp"
 #include "sparrow/types/data_type.hpp"
 #include "sparrow/utils/contracts.hpp"
 #include "sparrow/utils/iterator.hpp"
 #include "sparrow/utils/repeat_container.hpp"
+#include "sparrow/buffer/dynamic_bitset/dynamic_bitset.hpp"
+#include "sparrow/layout/layout_utils.hpp"
 
 namespace sparrow
 {
-    template <std::ranges::sized_range T, class CR, layout_offset OT = std::int32_t>
-    class variable_size_binary_array;
 
+    namespace detail
+    {   
+        template<class T, class OT>
+        struct variable_size_binary_format;
+
+        template<> 
+        struct variable_size_binary_format<std::string, std::int32_t> 
+        {
+            static std::string format(){return "u";}
+        };
+
+        template<>
+        struct variable_size_binary_format<std::string, std::int64_t>
+        {
+            static std::string format(){return "U";}
+        };
+
+        template<> 
+        struct variable_size_binary_format<std::vector<std::byte>, std::int32_t> 
+        {   
+            static std::string format(){return "z";}
+        };
+
+        template<>
+        struct variable_size_binary_format<std::vector<std::byte>, std::int64_t>
+        {
+            static std::string format(){return "Z";}
+        };
+
+        template<class T>
+        concept char_like = std::same_as<T, char> || std::same_as<T, std::byte> || std::same_as<T, std::uint8_t>;
+    }    
+
+
+
+    template <std::ranges::sized_range T, class CR, layout_offset OT>
+    class variable_size_binary_array_impl;
+
+    using string_array =     variable_size_binary_array_impl<std::string, std::string_view, std::int32_t>;
+    using big_string_array = variable_size_binary_array_impl<std::string, std::string_view, std::int64_t>;
```

Comment:
We should also define binary_array and big_binary_array ? In a future PR ?

Reply:
I felt like this was better for a follow up pr